joint-discovery-gatk4.hg38.wgs.inputs.json

```diff
@@ -1,92 +1,96 @@
+  "JointGenotyping.axiomPoly_resource_vcf_index": "gs://broad-references/hg38/v0/Axiom_Exome_Plus.genotypes.all_populations.poly.hg38.vcf.gz.tbi",
+  "JointGenotyping.hapmap_resource_vcf": "gs://broad-references/hg38/v0/hapmap_3.3.hg38.vcf.gz",
+  "JointGenotyping.hapmap_resource_vcf_index": "gs://broad-references/hg38/v0/hapmap_3.3.hg38.vcf.gz.tbi",
 
   "##_COMMENT5": "VQSR PARAMETERS",
-  "JointDiscovery_GATK4.BuildVQSRModelForSNPs.maxGaussians": 6,
-  "JointDiscovery_GATK4.BuildVQSRModelForINDELs.maxGaussians": 4,
-  "JointDiscovery_GATK4.SNP_annotations": ["DP", "QD", "FS", "SOR", "MQ", "MQRankSum", "ReadPosRankSum"],
-  "JointDiscovery_GATK4.INDEL_annotations": ["QD", "FS", "SOR", "MQRankSum", "ReadPosRankSum"],
-  "JointDiscovery_GATK4.SNP_filter_level": 99.7,
-  "JointDiscovery_GATK4.INDEL_filter_level": 99.7,
-  "JointDiscovery_GATK4.SNP_tranches": [100.0, 99.99, 99.95, 99.9, 99.8, 99.7, 99.6, 99.5, 99.4, 99.3, 99.2, 99.1, 99.0, 98.0, 97.0, 96.0, 95.0, 90.0],
-  "JointDiscovery_GATK4.INDEL_tranches": [100.0, 99.99, 99.95, 99.9, 99.8, 99.7, 99.6, 99.5, 99.0, 98.0, 97.0, 96.0, 95.0, 94.0, 93.0, 92.0, 91.0, 90.0],
+  "JointGenotyping.SNP_VQSR_downsampleFactor": 10,
+  "JointGenotyping.snp_filter_level": 99.7,
+  "JointGenotyping.indel_filter_level": 99.7,
+  "JointGenotyping.indel_recalibration_annotation_values": ["FS", "ReadPosRankSum", "MQRankSum", "QD", "SOR", "DP"],
+  "JointGenotyping.indel_recalibration_tranche_values": ["100.0", "99.95", "99.9", "99.5", "99.0", "97.0", "96.0", "95.0", "94.0", "93.5", "93.0", "92.0", "91.0", "90.0"],
+  "JointGenotyping.snp_recalibration_tranche_values": ["100.0", "99.95", "99.9", "99.8", "99.6", "99.5", "99.4", "99.3", "99.0", "98.0", "97.0", "90.0" ],
+  "JointGenotyping.snp_recalibration_annotation_values": ["QD", "MQRankSum", "ReadPosRankSum", "FS", "MQ", "SOR", "DP"],
 
   "##_COMMENT4": "DOCKERS", 
-  "JointDiscovery_GATK4.picard_docker": "vdauwera/picard:2.13.2",
-  "JointDiscovery_GATK4.gatk_docker": "broadinstitute/gatk:4.beta.5",
+  "JointGenotyping.python_docker": "python:2.7",
+  "JointGenotyping.gatk_docker": "broadinstitute/gatk:4.beta.6",
 
   "##_COMMENT5": "PATHS", 
-  "JointDiscovery_GATK4.gatk_launch_path": "/gatk/",
-  "JointDiscovery_GATK4.picard_path": "/usr/",
+  "JointGenotyping.gatk_path": "/gatk/gatk-launch",
 
   "##_COMMENT6": "JAVA OPTIONS", 
-  "JointDiscovery_GATK4.ImportGVCFs.java_opt": "-Xmx4g -Xms4g",
-  "JointDiscovery_GATK4.GenotypeGVCFs.java_opt": "-Xms8000m",
-  "JointDiscovery_GATK4.MergeVCFs.java_opt": "-Xms2000m",
-  "JointDiscovery_GATK4.BuildVQSRModelForSNPs.java_opt": "-Xms8000m",
-  "JointDiscovery_GATK4.BuildVQSRModelForINDELs.java_opt": "-Xms8000m",
-  "JointDiscovery_GATK4.ApplyRecalibrationFilterForSNPs.java_opt": "-Xms4000m",
-  "JointDiscovery_GATK4.ApplyRecalibrationFilterForINDELs.java_opt": "-Xms4000m",
+  "JointGenotyping.SNPsVariantRecalibratorScattered.java_opt": "-Xmx3g -Xms3g",
```

Comment:
I think if you want to give a user the ability to set the memory its better one by using optional values and putting defaults in the task like how it is done in the mutect wdl so they only have to optionally provide one value to any given call - https://github.com/broadinstitute/gatk/blob/master/scripts/mutect2_wdl/unsupported/mutect2_opt.wdl#L431.  If you want them to have full control over java opts then I would keep it as is.

Reply:
Not sure how much of a value there is in giving users ability to fine tune their memory setting,  you may have a better idea how often a user may need to change the memory. But i don't mind making the edit to keep things consistent with mutect.

Reply:
oh my point was that if this is expected to be passed in to the inputs json, isnt it also expected to be given as part of the method config in firecloud?  If a user doesnt have to make this a part of their method config then yea its fine as is.

Reply:
Resolved: Users will use the json's default values and have the option to change it if necessary.